packages/babel-parser/src/parser/expression.js

```diff
@@ -655,6 +655,15 @@ export default class ExpressionParser extends LValParser {
       node.object = base;
       node.property = this.parseMaybePrivateName();
       node.computed = false;
+      if (
+        node.property.type === "PrivateName" &&
+        node.object.type === "Super"
+      ) {
+        this.raise(
+          startPos,
+          "super is not allowed to be called on a private identifier of a class",
```

Comment:
Well, in this case `super` is not being called (it's not `super(#x)`).
What about `Private fields can't be accessed on super?`

Reply:
Yes that message makes more sense. I will make the changes accordingly and also update the tests. Thank you